src/main/java/org/zalando/nakadi/service/subscription/state/PartitionData.java

```diff
@@ -52,25 +53,58 @@
         this.commitOffset = commitOffset;
         this.sentOffset = commitOffset;
         this.lastSendMillis = currentTime;
+        this.batchWindowStartTimestamp = 0L;
     }
 
     @Nullable
-    List<ConsumedEvent> takeEventsToStream(final long currentTimeMillis, final int batchSize,
-                                           final long batchTimeoutMillis, final boolean streamTimeoutReached) {
+    List<ConsumedEvent> takeEventsToStream(final long currentTimeMillis, final long batchTimespanMillis,
+                                           final int batchSize, final long batchTimeoutMillis,
+                                           final boolean streamTimeoutReached) {
         final boolean countReached = (nakadiEvents.size() >= batchSize) && batchSize > 0;
         final boolean timeReached = (currentTimeMillis - lastSendMillis) >= batchTimeoutMillis;
-        if (countReached || timeReached) {
+        final long lastRecordTimestamp = lastRecordTimestamp();
```

Comment:
I think it makes sense to move `lastRecordTimestamp` and `batchWindowStartTimestamp` inside if clause, because it is used only there

Reply:
I'm not sure about this. About `lastRecordTimestamp` it is used twice in the condition on line 71 and `batchWindowStartTimestamp` is part of the state of this object.

Reply:
e8e2069b